### PR TITLE
adding an implementation of 'init' into Lwt module

### PR DESCRIPTION
**Pull request by Thibaut-Gudin**

### CODE REVIEW

**Comment by jrochel** [comment]
À nettoyer.

**Comment by Thibaut-Gudin** [comment]
Ça devrai être prêt à merger maintenant.

File: README.md [hunk 1]
```diff
@@ -64,6 +64,17 @@ Mixpanel.init ~token ~config ~name
 ```
   [See the default config options](https://github.com/mixpanel/mixpanel-js/blob/8b2e1f7b/src/mixpanel-core.js#L87-L110)
 
+### Mixpanel.Lwt.init
```
Comment (ilankri) [inline README.md]:
```suggestion
### `Mixpanel.Lwt.init`
```

File: README.md [hunk 2]
```diff
@@ -64,6 +64,17 @@ Mixpanel.init ~token ~config ~name
...
+The module "Lwt" provide an implementation of "init" that put your
+application in "wait" status until the "init" is over.
```
Comment (ilankri) [inline README.md]:
```suggestion
The module `Lwt` provides an implementation of `init` that puts your
application in "wait" status until the initialization is over.
```

File: src/mixpanel_lwt.mli [hunk 3]
```diff
@@ -6,3 +6,31 @@ val track :
...
+  ?cookie_expiration:int ->
+  ?upgrade:bool ->
+  ?disable_persistence:bool ->
+  ?disable_cookie:bool ->
+  ?secure_cookie:bool ->
+  ?ip:bool ->
+  ?property_blacklist:string list ->
+  ?name:string ->
+  unit ->
+  bool Lwt.t
```
Comment (ilankri) [inline src/mixpanel_lwt.mli]:
```suggestion
  unit Lwt.t
```

Since this function always returns `true`.

File: src/mixpanel_lwt.mli [hunk 4]
```diff
@@ -6,3 +6,31 @@ val track :
...
+  token:string ->
+  ?api_host:string ->
+  ?app_host:string ->
+  ?autotrack:bool ->
+  ?cdn:string ->
+  ?cross_subdomain_cookie:bool ->
+  ?persistence:string ->
+  ?persistence_name:string ->
+  ?cookie_name:string ->
+  ?loaded:unit ->
```
Comment (ilankri) [inline src/mixpanel_lwt.mli]:
```suggestion
```

This argument does not make sense for the promise-based interface.

File: src/mixpanel_lwt.mli (hunk 3 above)
Comment (Thibaut-Gudin) [reply src/mixpanel_lwt.mli]: But the code on "bs" works with a "Lwt.pick" in "bs_mixpanel.eliom" (https://github.com/besport/bs/pull/12543/files), did I have to change it too?

Comment (ilankri) [reply src/mixpanel_lwt.mli]: Yes, you need to the adjust the code accordingly.

Comment (Thibaut-Gudin) [reply src/mixpanel_lwt.mli]: I tried to change the type of the returned value but it failed, the compilation deduce that the typed returned is "bool" and not unit:
<<
 Values do not match:
         val init :
           token:string ->
           ?api_host:string ->
           ?app_host:string ->
           ?autotrack:bool ->
           ?cdn:string ->
           ?cross_subdomain_cookie:bool ->
           ?persistence:string ->
           ?persistence_name:string ->
           ?cookie_name:string ->
           ?loaded:bool ->
           ?store_google:bool ->
           ?save_referrer:bool ->
           ?test:bool ->
           ?verbose:bool ->
           ?img:bool ->
           ?track_links_timeout:int ->
           ?cookie_expiration:int ->
           ?upgrade:bool ->
           ?disable_persistence:bool ->
           ?disable_cookie:bool ->
           ?secure_cookie:bool ->
           ?ip:bool ->
           ?property_blacklist:string list ->
           ?name:string -> unit -> bool Lwt.t
       is not included in
         val init :
           token:string ->
           ?api_host:string ->
           ?app_host:string ->
           ?autotrack:bool ->
           ?cdn:string ->
           ?cross_subdomain_cookie:bool ->
           ?persistence:string ->
           ?persistence_name:string ->
           ?cookie_name:string ->
           ?store_google:bool ->
           ?save_referrer:bool ->
           ?test:bool ->
           ?verbose:bool ->
           ?img:bool ->
           ?track_links_timeout:int ->
           ?cookie_expiration:int ->
           ?upgrade:bool ->
           ?disable_persistence:bool ->
           ?disable_cookie:bool ->
           ?secure_cookie:bool ->
           ?ip:bool ->
           ?property_blacklist:string list ->
           ?name:string -> unit -> unit Lwt.t
>>
Maybe I haven't understand what the function have to return

Comment (ilankri) [reply src/mixpanel_lwt.mli]: It should work by adjusting the value passed as argument `loaded` to `Base.config`.

File: src/base.mli [hunk 5]
```diff
@@ -63,7 +63,7 @@ val config :
   ?persistence:string ->
   ?persistence_name:string ->
   ?cookie_name:string ->
-  ?loaded:(Ojs.t -> unit) ->
+  ?loaded:(unit -> unit) ->
```
Comment (ilankri) [inline src/base.mli:66]: It is a breaking change, so we should probably increase the major version number (i.e. move from 2.0 to 3.0).

Comment (Thibaut-Gudin) [reply src/base.mli:66]: I will have to change the number version of the "Group" module PR too but I'll do it (and the tags will be renamed too).

**Review by ilankri — APPROVED** [review]

**Comment by Thibaut-Gudin** [comment]
Je ne sais pas qui a la possibilité de pouvoir mergé cette PR de la librarie "Mixpanel" mais je suppose qu'il vaut mieux attendre que quelqu'un d'autre la merge je suppose?

**Comment by ilankri** [comment]
> Je ne sais pas qui a la possibilité de pouvoir mergé cette PR de la librarie "Mixpanel" mais je suppose qu'il vaut mieux attendre que quelqu'un d'autre la merge je suppose?

Oui, effectivement ! Je m'en charge de ce pas.

**Pull request merged**